polar-core/src/formatting.rs

```diff
@@ -106,6 +104,16 @@ pub fn format_params(args: &[Parameter], sep: &str) -> String {
         .join(sep)
 }
 
+/// Formats a vector of rules as a string-separated list.
+#[allow(clippy::ptr_arg)]
+pub fn format_rules(rules: &Rules, sep: &str) -> String {
```

Comment:
We could probably make this + `format_params` (above) generic over `&[T]` where `T: ToPolarString`, but it doesn't have to block this PR.